### DISCUSSION TITLE
A source member can map to multiple destination members

### DISCUSSION THREAD

**Discussion by Xriuk** [LuckyPennySoftware/AutoMapper#4002] [NA]
### Source/destination types

```csharp
class Order{
  public int ProductId {get; set;}
  public Product? Product {get; set;}
}

class OrderDto{
  public int ProductId {get; set;}
}
```

### Mapping configuration

```csharp
// ASP Net Core Dependency Injection - Global configuration
builder.Services.AddAutoMapper((serviceProvider, config) => {
  var postfixes = new[] { "Id", "Code" };
  config.RecognizePostfixes(postfixes);
  config.RecognizeDestinationPostfixes(postfixes);
  
  config.AllowNullCollections = true;
  config.AddCollectionMappers();
}, Assembly.GetExecutingAssembly());

// Actual map inside a profile
CreateMap<OrderDto, Order>();
```

### Expected behavior

I'd expect Automapper to match members with the same un-altered name first before proceeding to exclude pre/postfixes, so that ProductId of OrderDto gets mapped correctly to ProductId of Order.

### Actual behavior

I get an error saying missing mapping for int -> Product, because automapper tries to map Product(Id) to Product instead by removing the postfix.

**Comment by lbargaoanu** [comment]
Check [the execution plan](https://docs.automapper.org/en/latest/Understanding-your-mapping.html). You configured AM to map `ProductId` to both `ProductId` and `Product` and, as the validation says, it cannot do that. It seems that kind of config is not exactly a good idea :)

**Reply by Xriuk** [reply]
Yeah, that's kinda the point, the execution plan shouldn't be like that imo.
I think that AM should pick the first (and best) member if not explicitly configured (why would it map automatically to two members anyway?). I don't know how it's picking the members now in the code, so I don't know if it would be a big deal to change its behaviour or not.
